i3lock.c

```diff
@@ -632,6 +633,36 @@ void handle_screen_resize(void) {
     redraw_screen();
 }
 
+static bool verify_png_image(const char *image_path) {
+    if (!image_path) {
+        return false;
+    }
+
+    /* Check file exists and has correct PNG header */
+    FILE *png_file = fopen(image_path, "r");
+    if (png_file == NULL) {
+        fprintf(stderr, "Image file path \"%s\" cannot be opened: \"%s\"\n", image_path, strerror(errno));
+        return false;
+    }
+    unsigned char png_header[8];
+    memset(png_header, '\0', sizeof(png_header));
+    int bytes_read = fread(png_header, 1, sizeof(png_header), png_file);
+    fclose(png_file);
+    if (bytes_read != sizeof(png_header)) {
+        fprintf(stderr, "Could not read the size of PNG header in file \"%s\"\n", image_path);
```

Comment:
Sorry, missed this spot in the previous review, but let’s also make sure that this error message is as good as it can be. I suggest:

fprintf(stderr, "Could not read PNG header from \"%s\": %s\n", image_path, strerror(errno));

Reply:
Actually `fread` doesn't set `errno` to a valid value, I have "`No such file or directory`" which doesn't make sense. I propose to stick to:
```cpp
fprintf(stderr, "Could not read PNG header from \"%s\"\n", image_path);
```